vhdl_lang/src/ast/search.rs

```diff
@@ -1075,14 +1075,21 @@ impl Search for Declaration {
                 return_if_found!(open_info.search(ctx, searcher));
                 return_if_found!(file_name.search(ctx, searcher));
             }
-
             Declaration::Package(ref package_instance) => {
                 return_if_found!(package_instance.search(ctx, searcher));
             }
-
+            Declaration::PackageDeclaration(ref decl) => {
+                return_if_found!(decl.search(ctx, searcher));
+            }
+            Declaration::PackageBody(ref body) => {
+                return_if_found!(body.search(ctx, searcher));
+            }
             Declaration::Configuration(_) => {
                 // @TODO
             }
+            Declaration::Disconnection(_) => {
+                // @TODO
```

Comment:
This is used, for example, by the `find all references` functionality of the LSP. Here, you can just delegate the search method to each children of the `Disconnect` AST element. The special search methods in `ident`s or `expression`s take care of the rest.

Reply:
I have done it. But it doesn't work for some reason.

When I test it in:

```vhdl
library ieee;
use ieee.std_logic_1164.all;

entity ent is
end entity;

architecture arch of ent is
  signal foo,bar : std_logic := '0';
  constant abc : time := 10 ms;
  disconnect all : std_logic after abc;

begin

  foo <= bar after abc;

end arch;
```

... the "abc" is not listed when I use LSP goto references. Not quite sure why. Maybe because it must be a time_expression, not just a generic expression, mh.

Reply:
I can try to look at your branch later to understand what the problem could be. But there is nothing immediate where I see any issues.

Reply:
Ah, I think this will only work if you analyze the ident

Reply:
ok, I will try

Reply:
I don't know how but it seems I've got it. Although I must admit I tried to mimic what the other items did without fully understanding every step on the way. Better check if I didn't make an obvious mistake. Either way, my example above now works, very cool.

Reply:
All of what you have done looks really good to me. No obvious issues found.